### PR TITLE
Change an old JS rating plugin to a new and shiny one 

### PR DESCRIPTION
**Pull request by shiroginne**
Hello,

thank you for converting from spree and maintaining this gem, it has helped us a lot!

But the old js plugin for rating is not very 2017-ish :) so we've decided to change it to a new and shiny one - http://antenna.io/demo/jquery-bar-rating/examples/
Now the stars plugin is modern and contains several different themes. Additionally: you can apply different theme for every star rating in your application.
But there is one concern: it's not backward compatible. And it is a big change (so we've bumped gem version)

screen shot 2017-01-05 at 15 56 44

Is it possible to make such big changes? If it's not, that's fine we'll just override templates in our main app 🍻

### CODE REVIEW

**Comment by tvdeyen** [comment]
Thanks. We are ok with the newer rating plugin, but please remove all commits that are related to update this plugin to latest Solidus. These changes are handled by #6 

🍻

**Comment by shiroginne** [comment]
Hello,
sorry for delay with this PR. I've updated code with current master.

Thanks for considering to update the gem with breaking changes for current users.
I was not sure which version to choose, so I take the closest - '1.1.0'

File: solidus_reviews.gemspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Gem::Specification.new do |s|
   s.platform    = Gem::Platform::RUBY
   s.name        = 'solidus_reviews'
-  s.version     = '1.0.0'
+  s.version     = '1.1.0'
```
Comment (tvdeyen) [inline solidus_reviews.gemspec]: Please don't bump the version. We will do that.

File: Gemfile [hunk 2]
```diff
@@ -3,6 +3,7 @@ source 'https://rubygems.org'
 branch = ENV.fetch('SOLIDUS_BRANCH', 'master')
 gem "solidus", github: "solidusio/solidus", branch: branch
 gem "solidus_auth_devise", github: "solidusio/solidus_auth_devise"
+gem "jquery-bar-rating", git: "https://github.com/denkungsart/jquery-bar-rating.git"
```
Comment (tvdeyen) [inline Gemfile]: Please don't load the javascript via a gem; vendor it into the assets instead. Then stores can use their own rating plugin in the frontend if they want. 

Also, Gemfiles don't get loaded from engines. Dependencies have to be defined in the gemspec.

File: app/views/spree/feedback_reviews/_form.html.erb [hunk 3]
```diff
@@ -5,9 +5,11 @@
...
 <% end %>
 <%= Spree.t(:was_this_review_helpful) %>
-<% for i in 1..NB_STARS %>
-  <%= radio_button_tag "feedback_review[rating]",
-                       Spree.t('star', :count => i), false, :class => "star" %>
-<% end %>
+<%=
+  select_tag "feedback_review[rating]",
+  options_for_select((1..NB_STARS).to_a),
+  data: { "barrating-options": { theme: THEME } },
```
Comment (tvdeyen) [inline app/views/spree/feedback_reviews/_form.html.erb]: Please don't store the theme in a constant. Be explicit and set `theme: 'fontawesome-stars'`

File: app/assets/javascripts/spree/backend/solidus_reviews.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
-//= require jquery.rating
+//= require jquery-bar-rating
 //= require spree/backend
 
-// Navigating to a page with ratings via TurboLinks shows the radio buttons
-$(document).on('page:load', function () {
-  $('input[type=radio].star').rating();
+$(document).on('ready', function () {
+  $.each($("select.review-stars"), function(index, value) {
```
Comment (tvdeyen) [inline app/assets/javascripts/spree/backend/solidus_reviews.js]: Where in the backend views is this select? I couldn't find it.

Comment (shiroginne) [inline app/assets/javascripts/spree/backend/solidus_reviews.js]: To be honest, me too. This select is located in the **'spree/reviews/stars'** partial. I've just updated old JS files with new code

**Comment by shiroginne** [comment]
Sorry for long reply, I've updated the code

File: app/assets/javascripts/spree/backend/solidus_reviews.js (hunk 4 above)
Comment (tvdeyen) [inline app/assets/javascripts/spree/backend/solidus_reviews.js]: Could we just remove it then?

File: app/assets/stylesheets/spree/backend/solidus_reviews.css [hunk 5]
```diff
@@ -1,3 +1,4 @@
 /*
+ *= require jquery-bar-rating
```
Comment (tvdeyen) [inline app/assets/stylesheets/spree/backend/solidus_reviews.css]: I don't see a necessity for including that stylesheet in the the admin. We don't display rating stars in the backend.

**Comment by shiroginne** [comment]
Yes, we could remove rating from the backend. I've updated PR

File: app/assets/javascripts/spree/frontend/solidus_reviews.js [hunk 6]
```diff
@@ -1,7 +1,9 @@
-//= require jquery.rating
+//= require jquery.barrating
 //= require spree/frontend
 
-// Navigating to a page with ratings via TurboLinks shows the radio buttons
-$(document).on('page:load', function () {
-  $('input[type=radio].star').rating();
+$(document).on('ready', function () {
```
Comment (kennyadsl) [inline app/assets/javascripts/spree/frontend/solidus_reviews.js]: I think we can use `Spree.ready(function($) {` here.

File: app/views/spree/feedback_reviews/_form.html.erb [hunk 7]
```diff
@@ -5,9 +5,11 @@
...
 <% end %>
 <%= Spree.t(:was_this_review_helpful) %>
-<% for i in 1..NB_STARS %>
-  <%= radio_button_tag "feedback_review[rating]",
-                       Spree.t('star', :count => i), false, :class => "star" %>
-<% end %>
+<%=
+  select_tag "feedback_review[rating]",
+  options_for_select((1..NB_STARS).to_a),
+  data: { "barrating-options": { theme: "fontawesome-stars" } },
```
Comment (kennyadsl) [inline app/views/spree/feedback_reviews/_form.html.erb]: what about moving this `theme` option into an extension preference?

File: vendor/assets/stylesheets/jquery-bar-rating.css [hunk 8]
```diff
@@ -0,0 +1,4 @@
+/*
+  *= require_tree .
```
Comment (kennyadsl) [inline vendor/assets/stylesheets/jquery-bar-rating.css]: I'd move all themes into a dedicated directory and require that directory instead

**Comment by ericsaupe** [comment]
Closing as stale. Feel free to reopen!

**Pull request closed without merging**